File: codewars/reverse-string/reverse.js
function reverse(str) {
  var chars = str.split("");
  chars.forEach(function(e, i) {
    if(i < chars.length / 2) {
      rightIdx = chars.length - i - 1;
      chars[i] = chars[rightIdx];
      chars[rightIdx] = e;
    }
  });
  return chars.join("");
}

console.log(reverse("PALINDROMES"));

File: codewars/reverse-string/reverse.rb
def reverse(str)
  chars = str.chars
  chars.each_with_index do |e, i|
    if i < chars.count / 2 
      right_idx = chars.count - i - 1
      chars[i] = chars[right_idx]
      chars[right_idx] = e
    else
      break
    end 
  end
  chars.join
end
puts reverse("palindrome")

